docs/plugins/index.md

```diff
@@ -0,0 +1,296 @@
+  3. Your plugin will need to expose symbols with C-style function names. DOME cannot access functions whose names have been mangled.
+
+# Plugin Interfaces
+
+## Lifecycle hooks
+
+DOME can call specially named functions implemented by your plugin, at different times during the game loop. For this to work, you must ensure that your compiler does not mangle names.
+
+Returning any result other than `DOME_RESULT_SUCCESS` will cause DOME to abort and shutdown. You should use the [`log(text)`](#method-log) call to print an error before this.
+
+### Init
+
+```
+DOME_Result PLUGIN_onInit(DOME_getAPIFunction DOME_getAPI, DOME_Context ctx)
+```
+
+DOME calls this function when the plugin is loaded, which gives you a chance to perform any initialisation you need to.
+You can also signal to DOME that there was a problem by returning `DOME_RESULT_FAILURE`.
+
+This is also the best opportunity to acquire the available APIs, thanks to the `DOME_getAPI` function pointer, which is explained in the [API Services](#api-services) section. The structs returned from this call should be stored for use throughout the lifetime of your plugin.
+
+### Pre-Update
+```
+DOME_Result PLUGIN_preUpdate(DOME_Context ctx)
+```
+This hook is called before the Game.update step of the game loop.
+
+### Post-Update
+```
+DOME_Result PLUGIN_postUpdate(DOME_Context ctx)
+```
+This hook is called after the Game.update step of the game loop.
+
+### Pre-Draw
+```
+DOME_Result PLUGIN_preDraw(DOME_Context ctx)
+```
+
+This hook is called before the Game.draw step of the game loop.
+
+### Post-Draw
+```
+DOME_Result PLUGIN_postDraw(DOME_Context ctx)
+```
+This hook is called after the Game.draw step of the game loop.
+
+
+### Shutdown
+```
+DOME_Result PLUGIN_onShutdown(DOME_Context ctx)
+```
+This hook occurs when the plugin is being unloaded, usually because DOME is in the process of quitting. This is your last opportunity to free any resources your plugin is holding on to, and cleaning up any other background processes.
+
+
+
+# API Services
+
+The DOME Plugin API is split into different pieces, divided by general purpose and version. This is to allow maximum backwards-compatibility, as new features are added.
+The engine  will endeavour to support previous versions of an API for as long as possible, but no guarentees will be made for compatibility across major versions of DOME.
+
+APIs are provided as a struct of function pointers, returned from:
+```
+void* DOME_getAPI(API_TYPE type, int API_VERSION)
+```
+
+## Core
+
+This API allows your plugin to register modules and provides some basic utilities.
+
+### Acquisition
+
+```
+DOME_API_v0* core = (DOME_API_v0*)DOME_getAPI(API_DOME, DOME_API_VERSION);
+```
+
+### Enums: 
+
+#### enum: DOME_Result
+
+Various methods return an enum of type `DOME_Result`, which indicates success or failure. These are the valid values:
+
+ * `DOME_RESULT_SUCCESS`
+ * `DOME_RESULT_FAILURE`
+ * `DOME_RESULT_UNKNOWN`
```

Comment:
I think it's better to explain here what happens when this result is returned. It is stated in the example that any result other than `DOME_RESULT_SUCCESS` causes a fiber abort, but I think it's better to repeat that here for clarity.

Reply:
It's explained in the "Lifecycle Hooks" section. I'm not sure that it is worth repeating on the same page, especially when it may be used in other places which don't cause fiber aborts.